res/controllers/common-controller-scripts.js

```diff
@@ -28,6 +28,44 @@ String.prototype.toInt = function() {
     return a;
 };
 
+/**
+ * Discriminates whether an object was created using the `{}` synthax.
+ *
+ * Returns true when was an object was created using the `{}` synthax.
+ * False if the object is an instance of a class like Date or Proxy or an Array.
+ *
+ * Object.isSimpleObject({}) // true
+ * Object.isSimpleObject(null) // false
+ * Object.isSimpleObject(undefined) // false
+ * Object.isSimpleObject(new Date) // false
+ * Object.isSimpleObject(new (class {})()) // false
+ * @param {any} obj Object to test
+ * @returns {boolean} true if obj was created using the `{}` synthax, false otherwise
+ */
+Object.isSimpleObject = function(obj) {
+    return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+};
+
+/**
+ * Deeply merges 2 objects.
+ * @param source {object | Array} Object to merge target into
+ * @param target {object | Array} Object to merge into target
+ */
+Object.deepMerge = function(source, target) {
```

Comment:
I think the naming should be switched around. The keys of the second argument are merged into the first one, not the other way around. the jsdoc comment is a little nonsensical too IMO. The usage below in the componentJS ctor is then also switched around.